### PR TITLE
Correct id selection for WorkflowExecutionTerminated event.

### PR DESCRIPTION
**Pull request by robholland**
Fixes: NoMethodError: undefined method `scheduled_event_id' for #<Temporal::Api::History::V1::WorkflowExecutionTerminatedEventAttributes:0x000056370c436d38>

### CODE REVIEW

File: lib/temporal/workflow/history/event.rb [hunk 1]
```diff
@@ -10,8 +10,6 @@ class Event
           ACTIVITY_TASK_CANCELED
           TIMER_FIRED
           REQUEST_CANCEL_EXTERNAL_WORKFLOW_EXECUTION_FAILED
-          WORKFLOW_EXECUTION_SIGNALED
-          WORKFLOW_EXECUTION_TERMINATED
```
Comment (antstorm) [inline lib/temporal/workflow/history/event.rb:14]: Why were these removed?

Comment (robholland) [reply lib/temporal/workflow/history/event.rb:14]: The constant is only used as part of this case statement and those two event types don't have the `scheduled_event_id` unlike the others in that list so they don't feel like they belong.

Comment (robholland) [reply lib/temporal/workflow/history/event.rb:14]: It's fine for them to stay if you'd prefer now terminated event is correctly "shadowed" by the proceeding `when`. Just not really sure what determines which events should be in the list.

Comment (antstorm) [reply lib/temporal/workflow/history/event.rb:14]: Ah, I see, good point 👍

**Review by antstorm — APPROVED** [review]
